### PR TITLE
Echarts implementation

### PR DESCRIPTION
**Pull request by AmbossKeegan**
- Moved all BarCharts to echarts and removed old BarCharts
- Moved all HorizontalBarCharts to echarts and removed old HorizontalBarCharts
- Created the Sankey chart as a replacement for the chord graph
- Removed Visx and the Chord graph

Screen Shot 2023-06-05 at 8 09 11 PM

Screen Shot 2023-06-27 at 2 47 12 PM

### CODE REVIEW

File: src/client/src/components/chart/BarChartV2.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+  const seriesData = useMemo(() => {
+    if (data.length === 0) return { dates: [], series: [] };
+
+    const series = [
+      {
+        name: title,
+        type: 'bar',
+        emphasis: { focus: 'series' },
+        data: data.map((d: any) => d.Invoices),
```
Comment (AmbossKeegan) [inline src/client/src/components/chart/BarChartV2.tsx:52]: We will need to abstract this part out to the parent such that data of various kinds can be passed to the BarChart component.

**Review by secondl1ght — COMMENTED** [review]
Ok nice work, after my review I have a few suggestions/questions:

- we should probably change the minimum Y Axis interval to 1 because you can't have a fraction of a payment or invoice
- the time scale displayed on the X Axis is different from the original chart, not sure if this matters?
- it looks to me like the Payments chart is not working (I made a payment but nothing was displayed when I toggled the chart however it's working on master)
- there is an error in the console about server rendering (this is also on master but might be an opportunity to fix it now)
- the tooltip styling is different from the original chart (the original is much more minimal which I like better but it is personal preference)
- rather than trying to create a component for this Bar Chart we could just place the code inline in the desired file (like we talked about on the call today this approach is more flexible)
- the legend and main titles don't match the dropdown values for example Amount/Tokens vs. Count/Amount
- the previous chart did not have a Title or Legend, do we want to keep the new chart also minimal in appearance?

Please let me know if any of my comments are not clear, I can provide screenshots or more details if needed.

**Review by secondl1ght — COMMENTED** [review]
My comments are mostly nits at this point so I think it's ready for @apotdevin to take a final review now and see if he is happy with it.

- do we need a y label if its the same as the title?
- make title copy match dropdown copy for consistency
- ~is there a date format standard on TH like there is for Amboss? (for example to use in the tooltip dates)~
- we could format the volume numbers in the tooltip to make them more readable by adding commas (we could port the function over from Amboss)
- I like the colors/style of the original tooltip but again this is just a personal preference or nit

File: src/client/next.config.js [hunk 2]
```diff
@@ -10,6 +10,7 @@ module.exports = {
   reactStrictMode: true,
   poweredByHeader: false,
   basePath: process.env.BASE_PATH || '',
+  transpilePackages: ['echarts', 'echarts-for-react', 'zrender'],
```
Comment (apotdevin) [inline src/client/next.config.js:13]: Hadn't seen this config before. Why is it needed?

File: src/client/src/components/generic/helpers.tsx [hunk 3]
```diff
@@ -160,10 +160,11 @@ export const getPastFutureStr = (
 };
 
 export const getFormatDate = (
-  date: string | null | undefined
+  date: string | null | undefined,
+  style?: string
 ): string | null => {
   if (!date) return null;
-  return format(new Date(date), 'dd/MM/yyyy - HH:mm:ss');
+  return format(new Date(date), style || 'dd/MM/yyyy - HH:mm:ss');
```
Comment (apotdevin) [inline src/client/src/components/generic/helpers.tsx:167]: Let's use this example date for the default: `2022.02.24 8:32:21`

File: src/client/src/views/home/reports/flow/TransactionGraph.tsx [hunk 4]
```diff
@@ -48,12 +48,29 @@ export const TransactionsGraph: FC<TransactionsGraphProps> = ({
...
+  const { data: invoiceData, loading } = useGetInvoicesQuery();
   const { data: paymentsData, loading: paymentsLoading } =
     useGetPaymentsQuery();
 
+  const labels = useMemo(() => {
+    switch (type) {
+      case 'amount':
+        return {
+          yAxisLabel: `# of ${showPay ? 'Payments' : 'Invoices'}`,
+          title: `# of ${showPay ? 'Payments' : 'Invoices'}`,
```
Comment (apotdevin) [inline src/client/src/views/home/reports/flow/TransactionGraph.tsx:60]:
```suggestion
          yAxisLabel: `Amount of ${showPay ? 'Payments' : 'Invoices'}`,
          title: `Amount of ${showPay ? 'Payments' : 'Invoices'}`,
```

File: src/client/src/views/home/reports/flow/TransactionGraph.tsx [hunk 5]
```diff
@@ -102,10 +119,15 @@ export const TransactionsGraph: FC<TransactionsGraphProps> = ({
     );
   }
 
-  if (!data?.getInvoices.invoices.length) {
+  if (
+    (!showPay && !invoiceData?.getInvoices.invoices.length) ||
+    (showPay && !paymentsData?.getPayments.payments.length)
+  ) {
```
Comment (apotdevin) [inline src/client/src/views/home/reports/flow/TransactionGraph.tsx:125]: Nice!

**Review by apotdevin — COMMENTED** [review]
Let's also reduce the padding the chart has - seems to big right now

File: src/client/next.config.js (hunk 2 above)
Comment (AmbossKeegan) [reply src/client/next.config.js:13]: https://nextjs.org/docs/app/api-reference/next-config-js/transpilePackages

Speculation: It has something to do with the monorepo and an inability to pull and render components from those packages without transpiling in NextJS.

**Review by secondl1ght — CHANGES_REQUESTED** [review]
Nice, left some comments/suggestions inline!

**Comment by apotdevin** [comment]
- Padding around this chart is weird, lets try to center all charts inside the card.
image

- Same here. Can we center and add the x-axis. There also seems to be an extra empty category? 
image

- For the y-axis can we format numbers similar to amboss space? `1800000` -> `1.8m`
image

- This chart get's cut off 
image

**Comment by apotdevin** [comment]
What is needed to finish this PR @AmbossKeegan ?

**Comment by AmbossKeegan** [comment]
> What is needed to finish this PR @AmbossKeegan ?

Just pushed a commit with code that bounds the sankey data inside its chart. Also provided a mock-data function that can be used to generate an arbitrarily large sankey dataset for testing purposes.

As far as I am aware, this was the last outstanding piece of feedback that needed to be addressed for this PR.

**Pull request merged**